src/components/password/password.component.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { useRef, useState } from "react";
+import { TextboxProps } from "../textbox";
+import { StyledPassword, HiddenAriaLive } from "./password.style";
+import guid from "../../__internal__/utils/helpers/guid";
+import useLocale from "../../hooks/__internal__/useLocale";
+import ButtonMinor from "../button-minor/button-minor.component";
+
+export type TypeOptions = "text" | "password";
+
+export interface PasswordProps extends TextboxProps {
+  /** HTML type attribute of the input */
+  type?: TypeOptions;
+}
```

Comment:
Yeah tbh I'd be interested to see what we can do here, so `type` is not in `TextboxProps`, hence half the need to create a dedicated component, however by having `type` in the interface in theory we should be allowing consumers to manually change type on submission etc for security reasons. `StyledPassword` is just a `Textbox` which uses the `PasswordProps` interface, which lets us get around the `type` issue, if `type` wasn't used in this interface and we tried to pass it to `StyledPassword` it would throw an error, similar to how it would if we just rendered `Textbox` and it passed it to that.